### ISSUE TITLE
Unable to add Elastic search in Docksal

### ISSUE THREAD

**Issue by furija357** [docksal/docksal#1280]
Elastic search has been added into docksal.yml as per documentation https://docs.docksal.io/tools/elasticsearch/

After adding the service, on fin p start  I'm getting an error: 
Cannot extend service 'elasticsearch' in /././.docksal/stacks/services.yml: Service not found

version: "2.1"
services:
  elasticsearch:
    extends:
      file: ${HOME}/.docksal/stacks/services.yml
      service: elasticsearch

**Comment by achekulaev** [comment]
Post contents of your `${HOME}/.docksal/stacks/services.yml`

**Comment by furija357** [comment]
Output of fin config
```
---------------------
COMPOSE_PROJECT_NAME_SAFE: savvy
COMPOSE_FILE:
/home/bane/.docksal/stacks/volumes-bind.yml
/home/bane/.docksal/stacks/stack-default.yml
/var/www/html/savvy/.docksal/docksal.yml
ENV_FILE:
/var/www/html/savvy/.docksal/docksal.env

PROJECT_ROOT: /var/www/html/savvy
DOCROOT: docroot
VIRTUAL_HOST: savvy.docksal
VIRTUAL_HOST_ALIASES: *.savvy.docksal
IP: 192.168.64.100
MYSQL: 192.168.64.100:32779

Docker Compose configuration
---------------------
services:
  cli:
    dns:
    - 192.168.64.100
    - 8.8.8.8
    environment:
      BLACKFIRE_CLIENT_ID: null
      BLACKFIRE_CLIENT_TOKEN: null
      DOCROOT: docroot
      GIT_USER_EMAIL: branislav.radovanovic@etondigital.com
      GIT_USER_NAME: ''
      HOST_GID: null
      HOST_UID: null
      SECRET_ACAPI_EMAIL: null
      SECRET_ACAPI_KEY: null
      SECRET_PLATFORMSH_CLI_TOKEN: null
      SECRET_SSH_PRIVATE_KEY: null
      SECRET_TERMINUS_TOKEN: null
      VIRTUAL_HOST: savvy.docksal
      XDEBUG_CONFIG: remote_connect_back=0 remote_host=192.168.64.1
      XDEBUG_ENABLED: '0'
    hostname: cli
    image: docksal/cli:2.5-php7.1
    volumes:
    - docksal_ssh_agent:/.ssh-agent:ro
    - cli_home:/home/docker:rw
    - project_root:/var/www:rw,nocopy
  db:
    dns:
    - 192.168.64.100
    - 8.8.8.8
    environment:
      MYSQL_ALLOW_EMPTY_PASSWORD: null
      MYSQL_DATABASE: default
      MYSQL_INITDB_SKIP_TZINFO: null
      MYSQL_ONETIME_PASSWORD: null
      MYSQL_PASSWORD: user
      MYSQL_RANDOM_ROOT_PASSWORD: null
      MYSQL_ROOT_PASSWORD: root
      MYSQL_USER: user
    hostname: db
    image: docksal/db:1.2-mysql-5.6
    ports:
    - 0:3306/tcp
    volumes:
    - db_data:/var/lib/mysql:rw
    - project_root:/var/www:ro,nocopy
  mail:
    dns:
    - 192.168.64.100
    - 8.8.8.8
    hostname: mail
    image: mailhog/mailhog
    labels:
      io.docksal.cert-name: none
      io.docksal.virtual-host: mail.savvy.docksal,mail.savvy.docksal.*
      io.docksal.virtual-port: '8025'
    volumes:
    - project_root:/var/www:ro,nocopy
  web:
    depends_on:
      cli:
        condition: service_started
    dns:
    - 192.168.64.100
    - 8.8.8.8
    environment:
      APACHE_BASIC_AUTH_PASS: null
      APACHE_BASIC_AUTH_USER: null
      APACHE_DOCUMENTROOT: /var/www/docroot
    hostname: web
    image: docksal/web:2.1-apache2.4
    labels:
      io.docksal.cert-name: none
      io.docksal.permanent: "false"
      io.docksal.project-root: /var/www/html/savvy
      io.docksal.virtual-host: savvy.docksal,*.savvy.docksal,savvy.docksal.*
    volumes:
    - project_root:/var/www:ro,nocopy
version: '2.1'
volumes:
  cli_home: {}
  db_data: {}
  docksal_ssh_agent:
    external: true
    name: docksal_ssh_agent
  project_root:
    driver: local
    driver_opts:
      device: /var/www/html/savvy
      o: bind
      type: none

---------------------
```
output of fin sysinfo:

```
███  OS
Linux Ubuntu 16.04
Linux bane-B85M-HD3 4.4.0-171-generic #200-Ubuntu SMP Tue Dec 3 11:04:55 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux

███  ENVIRONMENT
MODE : Linux Kernel
DOCKER_HOST : 

███  FIN
fin version: 1.80.1

███  DOCKER COMPOSE
EXPECTED VERSION: 1.23.1
docker-compose version 1.21.2, build a133471
docker-py version: 3.3.0
CPython version: 3.6.5
OpenSSL version: OpenSSL 1.0.1t  3 May 2016

███  DOCKER
EXPECTED VERSION: 18.06.1-ce

Client:
 Version:           18.06.1-ce
 API version:       1.38
 Go version:        go1.10.3
 Git commit:        e68fc7a
 Built:             Tue Aug 21 17:24:51 2018
 OS/Arch:           linux/amd64
 Experimental:      false

Server:
 Engine:
  Version:          18.06.1-ce
  API version:      1.38 (minimum version 1.12)
  Go version:       go1.10.3
  Git commit:       e68fc7a
  Built:            Tue Aug 21 17:23:15 2018
  OS/Arch:          linux/amd64
  Experimental:     false
WARNING: No swap limit support

███  DOCKSAL: PROJECTS
project             STATUS              virtual host                                    project root
savvy               Up 2 hours          savvy.docksal,*.savvy.docksal,savvy.docksal.*   /var/www/html/savvy

███  DOCKSAL: VIRTUAL HOSTS
*.savvy.docksal
mail.savvy.docksal.*
mail.savvy.docksal
savvy.docksal.*
savvy.docksal

███  DOCKER: RUNNING CONTAINERS
CONTAINER ID        IMAGE                         COMMAND                  CREATED             STATUS                   PORTS                                                    NAMES
ff1ddf4e74f0        docksal/ssh-agent:1.2         "docker-entrypoint.s…"   2 hours ago         Up 2 hours (healthy)                                                              docksal-ssh-agent
41fbf4a08202        docksal/dns:1.1               "docker-entrypoint.s…"   2 hours ago         Up 2 hours (healthy)     192.168.64.100:53->53/udp                                docksal-dns
3df10cfc4e78        docksal/vhost-proxy:1.4       "docker-entrypoint.s…"   2 hours ago         Up 2 hours (unhealthy)   192.168.64.100:80->80/tcp, 192.168.64.100:443->443/tcp   docksal-vhost-proxy
03f5c791fe66        docksal/web:2.1-apache2.4     "httpd-foreground"       3 days ago          Up 2 hours               80/tcp, 443/tcp                                          savvy_web_1
bf4eaeb99198        docksal/cli:2.5-php7.1        "/opt/startup.sh sup…"   3 days ago          Up 2 hours (healthy)     22/tcp, 3000/tcp, 9000/tcp                               savvy_cli_1
7cec4d7987b6        mailhog/mailhog               "MailHog"                3 days ago          Up 2 hours               1025/tcp, 8025/tcp                                       savvy_mail_1
6dee8ba2c584        docksal/db:1.2-mysql-5.6      "/entrypoint.sh mysq…"   3 days ago          Up 2 hours               0.0.0.0:32779->3306/tcp                                  savvy_db_1
ec8cf88b9cb8        amazeeio/drupal:php70-basic   "/sbin/my_init"          3 weeks ago         Up 3 weeks               80/tcp, 443/tcp, 0.0.0.0:32771->3306/tcp                 verlagskv.docker.amazee.io
ca01e3c105c1        amazeeio/solr:5               "docker-entrypoint.s…"   3 weeks ago         Up 3 weeks               0.0.0.0:32770->8983/tcp                                  skv-website_solr_1

███  DOCKER: NETWORKS
NETWORK ID          NAME                     DRIVER              SCOPE
2f5772bcaf6e        _default                 bridge              local
d5cc68c85acb        amazeeio-network         bridge              local
18627ab585b2        bridge                   bridge              local
d16408f33b4d        cgoc_default             bridge              local
0cce878c56e5        drupal-starter_default   bridge              local
a70d4659c3d7        host                     host                local
e8283dcfe397        laca_default             bridge              local
481af6a6ac9e        none                     null                local
24b275ccbfc6        savvy_default            bridge              local

███  VIRTUALBOX
EXPECTED VERSION: 5.2.20
WARNING: The vboxdrv kernel module is not loaded. Either there is no module
         available for the current kernel (4.4.0-171-generic) or it failed to
         load. Please recompile the kernel module and install it by

           sudo /sbin/vboxconfig

         You will not be able to start VMs until this problem is fixed.
5.1.6r110634

███  HDD Usage
Filesystem      Size  Used Avail Use% Mounted on
udev            7,7G     0  7,7G   0% /dev
tmpfs           1,6G   74M  1,5G   5% /run
/dev/sda1       214G  137G   67G  68% /
tmpfs           7,7G  138M  7,6G   2% /dev/shm
tmpfs           5,0M  4,0K  5,0M   1% /run/lock
tmpfs           7,7G     0  7,7G   0% /sys/fs/cgroup
/dev/loop0      157M  157M     0 100% /snap/gnome-3-28-1804/110
/dev/loop3      238M  238M     0 100% /snap/wine-platform-i386/25
/dev/loop4      256K  256K     0 100% /snap/gtk2-common-themes/5
/dev/loop6       45M   45M     0 100% /snap/gtk-common-themes/1440
/dev/loop1      102M  102M     0 100% /snap/p7zip-desktop/220
/dev/loop8      228M  228M     0 100% /snap/wine-platform-i386/23
/dev/loop11      51M   51M     0 100% /snap/p7zip-desktop/163
/dev/loop12      74M   74M     0 100% /snap/wine-platform-3-stable/6
/dev/loop15     4,0M  4,0M     0 100% /snap/notepad-plus-plus/222
/dev/loop14      22M   22M     0 100% /snap/chromium-ffmpeg/13
/dev/loop13      55M   55M     0 100% /snap/core18/1650
/dev/loop16     228M  228M     0 100% /snap/wine-platform-i386/22
/dev/loop18      29M   29M     0 100% /snap/chromium-ffmpeg/15
/dev/loop19     161M  161M     0 100% /snap/gnome-3-28-1804/116
/dev/loop22      45M   45M     0 100% /snap/gtk-common-themes/1353
tmpfs           1,6G     0  1,6G   0% /run/user/123
tmpfs           1,6G  104K  1,6G   1% /run/user/1000
/dev/loop24     225M  225M     0 100% /snap/wine-platform-runtime/80
overlay         214G  137G   67G  68% /var/lib/docker/overlay2/4c5db91b373b6a5c8961a021a2350a4e5c3081bc27648e259a3073cdb7e91234/merged
shm              64M     0   64M   0% /var/lib/docker/containers/ca01e3c105c172a3067ece4a45b916c305d648ee5adb81da9ae4e32ba504d745/mounts/shm
overlay         214G  137G   67G  68% /var/lib/docker/overlay2/4262ebfb9fbcd657b9c2d9f0b0d0e81dd6cbd87182bf80fa538a1e786c3524bb/merged
shm              64M     0   64M   0% /var/lib/docker/containers/ec8cf88b9cb87ee870fcf2cbb673f77e2a345f27bff79704c0b6562d7c5ed96d/mounts/shm
/dev/loop23      55M   55M     0 100% /snap/core18/1668
/dev/loop5      4,0M  4,0M     0 100% /snap/notepad-plus-plus/223
/dev/loop7      256K  256K     0 100% /snap/gtk2-common-themes/9
/dev/loop2      142M  142M     0 100% /snap/opera/64
/dev/loop10      92M   92M     0 100% /snap/core/8592
/dev/loop9      225M  225M     0 100% /snap/wine-platform-runtime/89
/dev/loop20     142M  142M     0 100% /snap/opera/65
/dev/loop17      92M   92M     0 100% /snap/core/8689
overlay         214G  137G   67G  68% /var/lib/docker/overlay2/2a64eaa77f1b8a8a2fa662ece6beee904c2de5dbcbb66b8fb8b9030b8ba38ce8/merged
shm              64M     0   64M   0% /var/lib/docker/containers/3df10cfc4e785f231326f2a88f38a0cc2d4185afcd1e4cde79a1008bcbc349c2/mounts/shm
overlay         214G  137G   67G  68% /var/lib/docker/overlay2/03e10bb00d38994d6fb7d19b044d11b97c635d14bc2530270207288fead0f463/merged
shm              64M     0   64M   0% /var/lib/docker/containers/41fbf4a08202c88263858e46445fbcee051617efb3c9a568483ce9edc6b51dee/mounts/shm
overlay         214G  137G   67G  68% /var/lib/docker/overlay2/072fd731e8ec43eabb58a5be37872ea9ee62b98b3d2649219bb2e73c1d116af4/merged
shm              64M     0   64M   0% /var/lib/docker/containers/ff1ddf4e74f0259c62c16d463f7677d26ae85832e968db1224ddec6c13d3504d/mounts/shm
overlay         214G  137G   67G  68% /var/lib/docker/overlay2/32467f88315bcd2f9b94f10b8e55b076507d34c2ce31d914f5ecaf011f073946/merged
shm              64M     0   64M   0% /var/lib/docker/containers/03f5c791fe668462716fd462ef0e551f9675b9ac1b527abda18671eee72677bf/mounts/shm
overlay         214G  137G   67G  68% /var/lib/docker/overlay2/92d8e264220e66ec3cc988ed1a20223ea158fd7f2a0914183eddfb642bab60e3/merged
shm              64M     0   64M   0% /var/lib/docker/containers/bf4eaeb991984b286e911928c37a2754a48b312734eb86d5bcfaa334128ee751/mounts/shm
overlay         214G  137G   67G  68% /var/lib/docker/overlay2/b86a3e768aa36bee6e0d9de01e55fdc99dda4cc2796402578f81e5667b5b17a3/merged
shm              64M     0   64M   0% /var/lib/docker/containers/7cec4d7987b64644e161b0c3e212acbcf4373928ec075673a613677def8af4cb/mounts/shm
overlay         214G  137G   67G  68% /var/lib/docker/overlay2/f8c5a21362905b3c00fecbb0dfc5a83f00e55e3e6c83a52c636f4166310ed0de/merged
shm              64M     0   64M   0% /var/lib/docker/containers/6dee8ba2c584bde4058bbc97c145934b1350dcacd3f3f202e2284c5451224e7c/mounts/shm
```
output of cat ${HOME}/.docksal/stacks/services.yml
```

# This is a library of preconfigured services for Docksal.
# To use any service extend it from this file located at ${HOME}/.docksal/stacks/services.yml.
# See ${HOME}/.docksal/stacks/default.yml for a basic LAMP stack extending from web, db and cli services in this file.

version: "2.1"

services:
  # Web
  web:
    hostname: web
    image: ${WEB_IMAGE:-docksal/web:2.1-apache2.4}
    volumes:
      - project_root:/var/www:ro,nocopy  # Project root volume (read-only)
    labels:
      - io.docksal.virtual-host=${VIRTUAL_HOST},*.${VIRTUAL_HOST},${VIRTUAL_HOST}.*
      - io.docksal.cert-name=${VIRTUAL_HOST_CERT_NAME:-none}
      - io.docksal.project-root=${PROJECT_ROOT}
      - io.docksal.permanent=${SANDBOX_PERMANENT:-false}
    environment:
      - APACHE_DOCUMENTROOT=/var/www/${DOCROOT:-docroot}
      - APACHE_BASIC_AUTH_USER
      - APACHE_BASIC_AUTH_PASS
    dns:
      - ${DOCKSAL_DNS1}
      - ${DOCKSAL_DNS2}

  # DB (legacy)
  # TODO: Deprecated and to be removed in October 2018
  db:
    hostname: db
    image: ${DB_IMAGE:-docksal/db:1.2-mysql-5.6}
    ports:
      - "${MYSQL_PORT_MAPPING:-3306}"
    volumes:
      - project_root:/var/www:ro,nocopy  # Project root volume (read-only)
      - db_data:/var/lib/mysql  # Database data volume
    environment:
      - MYSQL_ROOT_PASSWORD=${MYSQL_ROOT_PASSWORD:-root}
      - MYSQL_USER=${MYSQL_USER:-user}
      - MYSQL_PASSWORD=${MYSQL_PASSWORD:-user}
      - MYSQL_DATABASE=${MYSQL_DATABASE:-default}
    dns:
      - ${DOCKSAL_DNS1}
      - ${DOCKSAL_DNS2}

  # DB: MySQL
  mysql:
    hostname: db
    image: ${DB_IMAGE:-docksal/db:1.2-mysql-5.6}
    ports:
      - "${MYSQL_PORT_MAPPING:-3306}"
    volumes:
      - project_root:/var/www:ro,nocopy  # Project root volume (read-only)
      - db_data:/var/lib/mysql  # Database data volume
    environment:
      - MYSQL_ROOT_PASSWORD=${MYSQL_ROOT_PASSWORD:-root}
      - MYSQL_USER=${MYSQL_USER:-user}
      - MYSQL_PASSWORD=${MYSQL_PASSWORD:-user}
      - MYSQL_DATABASE=${MYSQL_DATABASE:-default}
      - MYSQL_ALLOW_EMPTY_PASSWORD
      - MYSQL_RANDOM_ROOT_PASSWORD
      - MYSQL_ONETIME_PASSWORD
      - MYSQL_INITDB_SKIP_TZINFO
    dns:
      - ${DOCKSAL_DNS1}
      - ${DOCKSAL_DNS2}

  # DB: PostgreSQL
  pgsql:
    hostname: db
    image: ${DB_IMAGE:-postgres:9.6-alpine}
    ports:
      - "${PGSQL_PORT_MAPPING:-5432}"
    volumes:
      - project_root:/var/www:ro,nocopy  # Project root volume (read-only)
      - db_data:/var/lib/postgresql/data  # Database data volume
    environment:
      - POSTGRES_DB=${POSTGRES_DB:-default}
      - POSTGRES_USER=${POSTGRES_USER:-user}
      - POSTGRES_PASSWORD=${POSTGRES_PASSWORD:-user}
    dns:
      - ${DOCKSAL_DNS1}
      - ${DOCKSAL_DNS2}

  # CLI - Used for all console commands and tools.
  cli:
    hostname: cli
    image: ${CLI_IMAGE:-docksal/cli:2.5-php7.1}
    volumes:
      - project_root:/var/www:rw,nocopy  # Project root volume
      - docksal_ssh_agent:/.ssh-agent:ro  # Shared ssh-agent socket
      - cli_home:/home/docker  # Write-heavy directories should be in volumes. See https://github.com/docksal/docksal/issues/325
    environment:
      - GIT_USER_EMAIL
      - GIT_USER_NAME
      - HOST_UID
      - HOST_GID
      - DOCROOT
      - VIRTUAL_HOST
      - XDEBUG_ENABLED=${XDEBUG_ENABLED:-0}
      - XDEBUG_CONFIG=remote_connect_back=0 remote_host=${DOCKSAL_HOST_IP}  # Point xdebug to the host IP
      - BLACKFIRE_CLIENT_ID
      - BLACKFIRE_CLIENT_TOKEN
      - SECRET_SSH_PRIVATE_KEY
      - SECRET_ACAPI_EMAIL
      - SECRET_ACAPI_KEY
      - SECRET_PLATFORMSH_CLI_TOKEN
      - SECRET_TERMINUS_TOKEN
    dns:
      - ${DOCKSAL_DNS1}
      - ${DOCKSAL_DNS2}

  # Varnish
  varnish:
    hostname: varnish
    image: ${VARNISH_IMAGE:-docksal/varnish:1.1-varnish5}
    volumes:
      - project_root:/var/www:ro,nocopy  # Project root volume (read-only)
    labels:
      - io.docksal.virtual-host=varnish.${VIRTUAL_HOST},varnish.${VIRTUAL_HOST}.*
      - io.docksal.cert-name=${VIRTUAL_HOST_CERT_NAME:-none}
    environment:
      - VARNISH_BACKEND_HOST=web
    dns:
      - ${DOCKSAL_DNS1}
      - ${DOCKSAL_DNS2}

  # Memcached
  memcached:
    hostname: memcached
    image: ${MEMCACHED_IMAGE:-memcached:1.4-alpine}
    command: ["-m", "128"]  # Set memcached memory limit to 128 MB by default
    dns:
      - ${DOCKSAL_DNS1}
      - ${DOCKSAL_DNS2}

  # Solr
  solr:
    hostname: solr
    image: ${SOLR_IMAGE:-docksal/solr:1.0-solr4}
    volumes:
      - project_root:/var/www:ro,nocopy  # Project root volume (read-only)
    labels:
      - io.docksal.virtual-host=solr.${VIRTUAL_HOST},solr.${VIRTUAL_HOST}.*
      - io.docksal.virtual-port=8983
      - io.docksal.cert-name=${VIRTUAL_HOST_CERT_NAME:-none}
    dns:
      - ${DOCKSAL_DNS1}
      - ${DOCKSAL_DNS2}

  # MailHog
  mail:
    hostname: mail
    image: ${MAILHOG_IMAGE:-mailhog/mailhog}
    volumes:
      - project_root:/var/www:ro,nocopy  # Project root volume (read-only)
    labels:
      - io.docksal.virtual-host=mail.${VIRTUAL_HOST},mail.${VIRTUAL_HOST}.*
      - io.docksal.virtual-port=8025
      - io.docksal.cert-name=${VIRTUAL_HOST_CERT_NAME:-none}
    dns:
      - ${DOCKSAL_DNS1}
      - ${DOCKSAL_DNS2}

  # Blackfire
  blackfire:
    image: blackfire/blackfire
    environment:
      - BLACKFIRE_SERVER_ID
      - BLACKFIRE_SERVER_TOKEN

```

**Comment by achekulaev** [comment]
Your Docksal version is too outdated
